examples/user_guide/Network_Graphs.ipynb

```diff
@@ -0,0 +1,335 @@
+    "nodes = hv.Nodes((x, y, node_indices, ['Output']+['Input']*(N-1)),  vdims=['Type'])\n",
+    "hv.Graph(((source, target), nodes, paths)).redim.range(**padding)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Working with NetworkX"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NetworkX is a very useful library when working with network graphs and the Graph Element provides ways of importing a NetworkX Graph directly. Here we will load the Karate Club graph and use the ``circular_layout`` function provided by NetworkX to lay it out:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph [tools=['hover']]\n",
+    "import networkx as nx\n",
+    "G = nx.karate_club_graph()\n",
+    "hv.Graph.from_networkx(G, nx.layout.circular_layout).redim.range(**padding)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Animating graphs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph {+framewise}\n",
+    "layouts = [nx.layout.circular_layout, nx.layout.fruchterman_reingold_layout, nx.layout.spectral_layout,\n",
+    "           nx.layout.spring_layout, nx.layout.random_layout]\n",
+    "\n",
+    "G = nx.karate_club_graph()\n",
+    "hv.HoloMap({l.__name__: hv.Graph.from_networkx(G, l) for l in layouts},\n",
+    "           kdims=['Layout'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Real world graphs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a final example let's look at a slightly larger graph. We will load a dataset of a Facebook network consisting a number of friendship groups identified by their ``'circle'``. We will load the edge and node data using pandas and then color each node by their friendship group using many of the things we learned above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%opts Nodes Graph [width=800 height=800 xaxis=None yaxis=None tools=['hover']]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph [color_index='circle']\n",
+    "%%opts Graph (node_size=10 edge_line_width=1)\n",
+    "colors = ['#000000']+hv.Cycle('Category20').values\n",
+    "edges_df = pd.read_csv('../assets/fb_edges.csv')\n",
+    "fb_nodes = hv.Nodes(pd.read_csv('../assets/fb_nodes.csv')).sort()\n",
+    "fb_graph = hv.Graph((edges_df, fb_nodes), label='Facebook Circles')\n",
+    "fb_graph = fb_graph.redim.range(x=(-0.05, 1.05), y=(-0.05, 1.05)).opts(style=dict(cmap=colors))\n",
+    "fb_graph"
+   ]
+  },
```

Comment:
I think we need to work on the visibility of the hovering.  Right now, it's clear that nodes and edges are being selected, because they change color, but as soon as the mouse stops moving it is very difficult to see the difference between selected and unselected nodes and especially edges:
![image](https://user-images.githubusercontent.com/1695496/30495119-56cf038e-9a10-11e7-9005-cf0edd182cac.png)

![image](https://user-images.githubusercontent.com/1695496/30495125-6118f8a4-9a10-11e7-9833-44b785d9c994.png)

Here the selected node has a slightly different outline, which is very difficult to see, and the selected edges turn red, which is only visible because these happen to be on white; when they are in a complex nest it's super hard to make them out.

Can we make all the non-selected nodes and edges turn dim (Hunt's suggestion)?  Or make the selected ones be much thicker than the rest, so that they actually stand out?

Reply:
I'll look into it.

Reply:
Just checked, this is another thing to file with bokeh, the nonselection style options are not exposed at the moment.

Reply:
Ok, please do file those issues, and you can tag me if you need someone other than you to express a user-level opinion.  But none of those things should hold up this PR.

Reply:
I was wrong, can probably get this working.

Reply:
I realize now what the issue was, there is no way of styling something that's not being hovered over. I can only customize non-selected nodes/edges, e.g. when using the tap or box_select tools.

Reply:
Here's what that looks like:

![bokeh_plot](https://user-images.githubusercontent.com/1550771/30497028-334b440c-9a49-11e7-9983-88d83f942bd5.png)

Reply:
Nice!